virtio-devices/src/epoll_helper.rs

```diff
@@ -172,6 +173,76 @@ impl EpollHelper {
             }
         }
     }
+
+    #[cfg(fuzzing)]
+    // Require to have a 'queue_evt' being kicked before calling
+    // and return when no epoll events are active
+    pub fn run(
+        &mut self,
+        paused: Arc<AtomicBool>,
+        paused_sync: Arc<Barrier>,
+        handler: &mut dyn EpollHelperHandler,
+    ) -> std::result::Result<(), EpollHelperError> {
+        const EPOLL_EVENTS_LEN: usize = 100;
+        let mut events = vec![epoll::Event::new(epoll::Events::empty(), 0); EPOLL_EVENTS_LEN];
+
+        loop {
+            let num_events = match epoll::wait(self.epoll_file.as_raw_fd(), 0, &mut events[..]) {
+                Ok(res) => res,
+                Err(e) => {
+                    if e.kind() == std::io::ErrorKind::Interrupted {
+                        // It's well defined from the epoll_wait() syscall
+                        // documentation that the epoll loop can be interrupted
+                        // before any of the requested events occurred or the
+                        // timeout expired. In both those cases, epoll_wait()
+                        // returns an error of type EINTR, but this should not
+                        // be considered as a regular error. Instead it is more
+                        // appropriate to retry, by calling into epoll_wait().
+                        continue;
+                    }
+                    return Err(EpollHelperError::Wait(e));
+                }
+            };
+
+            // Return when no epoll events are active
+            if num_events == 0 {
+                return Ok(());
+            }
```

Comment:
I'm not sure with this design you'll be able to cover the `completion` event you're mentioning in the commit message. The problem I see is that you're gonna find the `queue` event first, but if it takes some time for the `completion` event to be triggered (think about asynchronous accesses with `io_uring`), the next iteration of the epoll loop will find no event at all and it will return without waiting for the `completion` event.

Reply:
Right. This PR is only about handling a sequence of synchronous events.  The example of virtio-block from the commit message is talking about the synchronous backends (e.g. `raw_sync`, `qcow_sync`, etc). I will make that commit message more specific.

As we spoke earlier, handling async events will be a totally different story and will out of the scope of this PR. The challenge is the conflict between the nature of async (that requires indeterministic waiting time) and the nature of fuzzing (that requires speed to be effective).

Reply:
Sounds good

Reply:
I think it's fine it tests sync only.